app/jobs/update_allele_registry_ids.rb

```diff
@@ -1,72 +1,34 @@
 require 'rbconfig'
 
-class UpdateAlleleRegistryIds < ActiveJob::Base
+class UpdateAlleleRegistryIds < AlleleRegistryIds
   def perform(recurring = true)
     begin
-      for v in Variant.where(allele_registry_id: nil) do
+      Variant.where.not(allele_registry_id: nil).each do |v|
+        old_allele_registry_id = v.allele_registry_id
         allele_registry_id = get_allele_registry_id(v)
-        v.allele_registry_id = allele_registry_id
-        v.save
-      end
-      for v in Variant.where.not(allele_registry_id: nil) do
-        allele_registry_id = get_allele_registry_id(v)
-        add_allele_registry_link(allele_registry_id)
+        if allele_registry_id != old_allele_registry_id
+          v.allele_registry_id = allele_registry_id
+          v.save
+          add_allele_registry_link(allele_registry_id)
+          #delete the linkout if no other variant has this allele registry ID
+          if Variant.find_by(allele_registry_id: old_allele_registry_id).blank?
```

Comment:
```suggestion
          if Variant.find_by(allele_registry_id: old_allele_registry_id).exists?
```

Reply:
This one is my bad for not explaining it better, sorry to create extra work!

`find_by` eagerly resolves; ie once you call it, you either have `nil` or your `Variant` instance, its not lazy like most ActiveRecord query methods, so if you stick with `find_by` then, yeah, you'll want to use `blank?` because you no longer have a query object, you just have your result. 

`where` on the other hand is lazy like the rest of the query methods, so calling `Variant.where(foo: bar)` returns an ActiveRecord query object that you can keep building on, and it only actually runs the query and resolves when you force it to (by iterating over the results, or calling `.all` or whatever). So in order to use `exists?` you'll want to switch to `.where` or you can just revert it all and its not a big deal either.

Reply:
Thanks for the explanation. This seems to be a thing I do a lot (loading the instances instead of working with the ActiveRecord objects). I will try to be more mindful of this in the future.